src/services/textChanges.ts

```diff
@@ -241,6 +249,13 @@ namespace ts.textChanges {
             return this;
         }
 
+        public replaceNodes(sourceFile: SourceFile, oldNodes: Node | Node[], newNodes: Node | Node[], options: ChangeNodeOptions & ChangeNodesOptions = {}) {
+            const startPosition = getAdjustedStartPosition(sourceFile, isArray(oldNodes) ? oldNodes[0] : oldNodes, options, Position.Start);
+            const endPosition = getAdjustedEndPosition(sourceFile, isArray(oldNodes) ? lastOrUndefined(oldNodes) : oldNodes, options);

```

Comment:
It's like the .NET version (actual signature is `export function lastOrUndefined<T>(array: T[]): T | undefined {`)